// sh0-0ta-poker-server/src/stores/memoryStore.js
import { v4 as uuid } from 'uuid';

export const users = new Map();
export const purchases = [];
export const sponsors = new Map();
export const tournaments = new Map();
export const leagues = new Map();
export const leaderboards = new Map();

export const cosmeticsCatalog = [
  // Avatars (Free)
  { id: 'avatar-gold-shooter', name: 'Gold Shooter', category: 'avatars', slot: 'avatar', priceCents: 0, rarity: 'common', icon: '🥇' },
  { id: 'avatar-royal-player', name: 'Royal Player', category: 'avatars', slot: 'avatar', priceCents: 0, rarity: 'common', icon: '👑' },
  { id: 'avatar-neon-ace', name: 'Neon Ace', category: 'avatars', slot: 'avatar', priceCents: 0, rarity: 'common', icon: '🎯' },
  { id: 'avatar-h-town-legend', name: 'H-Town Legend', category: 'avatars', slot: 'avatar', priceCents: 0, rarity: 'common', icon: '🤠' },
  { id: 'avatar-queen-of-clubs', name: 'Queen of Clubs', category: 'avatars', slot: 'avatar', priceCents: 0, rarity: 'common', icon: '🃛' },
  { id: 'avatar-spade-king', name: 'Spade King', category: 'avatars', slot: 'avatar', priceCents: 0, rarity: 'common', icon: '♠️' },
  { id: 'avatar-diamond-boss', name: 'Diamond Boss', category: 'avatars', slot: 'avatar', priceCents: 0, rarity: 'common', icon: '💎' },
  { id: 'avatar-club-captain', name: 'Club Captain', category: 'avatars', slot: 'avatar', priceCents: 0, rarity: 'common', icon: '♣️' },

  // Watches
  { id: 'watch-crowntime-presidential', name: 'CrownTime Presidential Gold', category: 'watches', slot: 'watch', priceCents: 1999, rarity: 'epic', icon: '⌚' },
  { id: 'watch-crowntime-diamond', name: 'CrownTime Diamond Jubilee', category: 'watches', slot: 'watch', priceCents: 2999, rarity: 'legendary', icon: '⌚' },
  { id: 'watch-midnight-legacy', name: 'Midnight Legacy Watch', category: 'watches', slot: 'watch', priceCents: 1499, rarity: 'rare', icon: '⌚' },
  { id: 'watch-rose-gold-mogul', name: 'Rose Gold Mogul Watch', category: 'watches', slot: 'watch', priceCents: 1899, rarity: 'epic', icon: '⌚' },
  { id: 'watch-iced-out-champion', name: 'Iced Out Champion Watch', category: 'watches', slot: 'watch', priceCents: 2499, rarity: 'legendary', icon: '⌚' },

  // Jewelry - Chains
  { id: 'chain-gold-rope', name: 'Gold Rope Chain', category: 'jewelry', slot: 'chain', priceCents: 999, rarity: 'rare', icon: '⛓️' },
  { id: 'chain-diamond-tennis', name: 'Diamond Tennis Chain', category: 'jewelry', slot: 'chain', priceCents: 1999, rarity: 'epic', icon: '⛓️' },
  { id: 'chain-black-excellence', name: 'Black Excellence Pendant', category: 'jewelry', slot: 'chain', priceCents: 2999, rarity: 'legendary', icon: '⛓️' },
  { id: 'chain-founder-medallion', name: 'Founder Medallion', category: 'jewelry', slot: 'chain', priceCents: 4999, rarity: 'founder', icon: '⛓️' },

  // Jewelry - Rings
  { id: 'ring-champion', name: 'Champion Ring', category: 'jewelry', slot: 'ring', priceCents: 1299, rarity: 'rare', icon: '💍' },
  { id: 'ring-diamond', name: 'Diamond Grillz', category: 'jewelry', slot: 'ring', priceCents: 1599, rarity: 'epic', icon: '💍' },
  { id: 'ring-gold-grillz', name: 'Gold Grillz', category: 'jewelry', slot: 'ring', priceCents: 899, rarity: 'common', icon: '💍' },

  // Vehicles
  { id: 'vehicle-phantom-luxe', name: 'Phantom Luxe Sedan', category: 'vehicles', slot: 'vehicle', priceCents: 1999, rarity: 'epic', icon: '🏎️' },
  { id: 'vehicle-neon-panther', name: 'Neon Panther Coupe', category: 'vehicles', slot: 'vehicle', priceCents: 1799, rarity: 'epic', icon: '🏎️' },
  { id: 'vehicle-candy-paint', name: 'Candy Paint Slab', category: 'vehicles', slot: 'vehicle', priceCents: 1599, rarity: 'rare', icon: '🏎️' },
  { id: 'vehicle-midnight-suv', name: 'Midnight Executive SUV', category: 'vehicles', slot: 'vehicle', priceCents: 1799, rarity: 'epic', icon: '🏎️' },
  { id: 'vehicle-royal-roadster', name: 'Royal Gold Roadster', category: 'vehicles', slot: 'vehicle', priceCents: 2499, rarity: 'legendary', icon: '🏎️' },
  { id: 'vehicle-h-town-drop-top', name: 'H-Town Drop Top', category: 'vehicles', slot: 'vehicle', priceCents: 1899, rarity: 'epic', icon: '🏎️' },
  { id: 'vehicle-pearl-coupe', name: 'Pearl White Luxury Coupe', category: 'vehicles', slot: 'vehicle', priceCents: 1699, rarity: 'rare', icon: '🏎️' },
  { id: 'vehicle-black-diamond-super', name: 'Black Diamond Supercar', category: 'vehicles', slot: 'vehicle', priceCents: 2999, rarity: 'legendary', icon: '🏎️' },

  // Homes
  { id: 'home-h-town-estate', name: 'H-Town Estate', category: 'homes', slot: 'home', priceCents: 2499, rarity: 'legendary', icon: '🏠' },
  { id: 'home-creator-mansion', name: 'Creator Mansion', category: 'homes', slot: 'home', priceCents: 2299, rarity: 'epic', icon: '🏠' },
  { id: 'home-penthouse-suite', name: 'Penthouse Suite', category: 'homes', slot: 'home', priceCents: 1999, rarity: 'epic', icon: '🏠' },
  { id: 'home-lakefront-legacy', name: 'Lakefront Legacy House', category: 'homes', slot: 'home', priceCents: 2199, rarity: 'epic', icon: '🏠' },
  { id: 'home-southside-loft', name: 'Southside Luxury Loft', category: 'homes', slot: 'home', priceCents: 1699, rarity: 'rare', icon: '🏠' },
  { id: 'home-studio-house', name: 'Studio House', category: 'homes', slot: 'home', priceCents: 1299, rarity: 'rare', icon: '🏠' },
  { id: 'home-royal-estate', name: 'Royal Family Estate', category: 'homes', slot: 'home', priceCents: 2699, rarity: 'legendary', icon: '🏠' },
  { id: 'home-vip-lounge', name: 'VIP Poker Lounge', category: 'homes', slot: 'home', priceCents: 1999, rarity: 'epic', icon: '🏠' },

  // Outfits
  { id: 'outfit-tailored-gold', name: 'Tailored Gold Suit', category: 'outfits', slot: 'outfit', priceCents: 1199, rarity: 'rare', icon: '👔' },
  { id: 'outfit-black-mogul', name: 'All Black Mogul Fit', category: 'outfits', slot: 'outfit', priceCents: 999, rarity: 'common', icon: '👔' },
  { id: 'outfit-sunday-best', name: 'Sunday Best Fit', category: 'outfits', slot: 'outfit', priceCents: 899, rarity: 'common', icon: '👔' },
  { id: 'outfit-hbcu-jacket', name: 'HBCU Homecoming Jacket', category: 'outfits', slot: 'outfit', priceCents: 1299, rarity: 'rare', icon: '👔' },
  { id: 'outfit-creator-hoodie', name: 'Creator Hoodie', category: 'outfits', slot: 'outfit', priceCents: 899, rarity: 'common', icon: '👔' },
  { id: 'outfit-velvet-lounge', name: 'Velvet Lounge Jacket', category: 'outfits', slot: 'outfit', priceCents: 1399, rarity: 'rare', icon: '👔' },
  { id: 'outfit-royal-tracksuit', name: 'Royal Purple Tracksuit', category: 'outfits', slot: 'outfit', priceCents: 1099, rarity: 'rare', icon: '👔' },
  { id: 'outfit-luxury-streetwear', name: 'Luxury Streetwear Set', category: 'outfits', slot: 'outfit', priceCents: 1499, rarity: 'epic', icon: '👔' },

  // Profile Backgrounds
  { id: 'background-hbcu-yard', name: 'HBCU Yard', category: 'backgrounds', slot: 'background', priceCents: 799, rarity: 'common', icon: '🌅' },
  { id: 'background-barbershop', name: 'Barbershop Talk', category: 'backgrounds', slot: 'background', priceCents: 999, rarity: 'rare', icon: '🌅' },
  { id: 'background-cookout', name: 'Sunday Cookout', category: 'backgrounds', slot: 'background', priceCents: 899, rarity: 'common', icon: '🌅' },
  { id: 'background-jazz-lounge', name: 'Jazz Lounge', category: 'backgrounds', slot: 'background', priceCents: 1199, rarity: 'rare', icon: '🌅' },
  { id: 'background-studio-session', name: 'Studio Session', category: 'backgrounds', slot: 'background', priceCents: 1099, rarity: 'rare', icon: '🌅' },
  { id: 'background-city-lights', name: 'City Lights', category: 'backgrounds', slot: 'background', priceCents: 899, rarity: 'common', icon: '🌅' },
  { id: 'background-juneteenth-gold', name: 'Juneteenth Gold', category: 'backgrounds', slot: 'background', priceCents: 1499, rarity: 'epic', icon: '🌅' },
  { id: 'background-wall-street', name: 'Black Wall Street Legacy', category: 'backgrounds', slot: 'background', priceCents: 1699, rarity: 'epic', icon: '🌅' },
  { id: 'background-southern-royalty', name: 'Southern Royalty', category: 'backgrounds', slot: 'background', priceCents: 1299, rarity: 'rare', icon: '🌅' },
  { id: 'background-houston-night', name: 'Houston Night Ride', category: 'backgrounds', slot: 'background', priceCents: 1399, rarity: 'rare', icon: '🌅' },

  // Card Backs
  { id: 'card-back-gold', name: 'Gold Card Back', category: 'cardBacks', slot: 'cardBack', priceCents: 499, rarity: 'common', icon: '🃏' },
  { id: 'card-back-black-diamond', name: 'Black Diamond Card Back', category: 'cardBacks', slot: 'cardBack', priceCents: 599, rarity: 'rare', icon: '🃏' },
  { id: 'card-back-neon-purple', name: 'Neon Purple Card Back', category: 'cardBacks', slot: 'cardBack', priceCents: 549, rarity: 'common', icon: '🃏' },
  { id: 'card-back-wall-street', name: 'Black Wall Street Card Back', category: 'cardBacks', slot: 'cardBack', priceCents: 699, rarity: 'rare', icon: '🃏' },
  { id: 'card-back-hbcu', name: 'HBCU Homecoming Back', category: 'cardBacks', slot: 'cardBack', priceCents: 749, rarity: 'epic', icon: '🃏' },

  // Table Skins
  { id: 'table-wall-street', name: 'Black Wall Street Table', category: 'tableSkins', slot: 'tableSkin', priceCents: 699, rarity: 'rare', icon: '🎰' },
  { id: 'table-hbcu', name: 'HBCU Homecoming Table', category: 'tableSkins', slot: 'tableSkin', priceCents: 749, rarity: 'epic', icon: '🎰' },
  { id: 'table-royal-gold', name: 'Royal Gold Felt', category: 'tableSkins', slot: 'tableSkin', priceCents: 699, rarity: 'rare', icon: '🎰' },
  { id: 'table-southern-nights', name: 'Southern Nights Felt', category: 'tableSkins', slot: 'tableSkin', priceCents: 649, rarity: 'common', icon: '🎰' },
  { id: 'table-jazz-club', name: 'Jazz Club Table', category: 'tableSkins', slot: 'tableSkin', priceCents: 799, rarity: 'epic', icon: '🎰' },
  { id: 'table-barbershop-lounge', name: 'Barbershop Lounge Table', category: 'tableSkins', slot: 'tableSkin', priceCents: 649, rarity: 'common', icon: '🎰' },
  { id: 'table-cookout-classic', name: 'Cookout Classic Table', category: 'tableSkins', slot: 'tableSkin', priceCents: 549, rarity: 'common', icon: '🎰' },
  { id: 'table-vip-penthouse', name: 'VIP Penthouse Table', category: 'tableSkins', slot: 'tableSkin', priceCents: 1499, rarity: 'epic', icon: '🎰' },
  { id: 'table-h-town', name: 'H-Town Table Skin', category: 'tableSkins', slot: 'tableSkin', priceCents: 399, rarity: 'common', icon: '🎰' },
  { id: 'table-midnight', name: 'Midnight Felt Table Skin', category: 'tableSkins', slot: 'tableSkin', priceCents: 499, rarity: 'common', icon: '🎰' },

  // Avatar Frames
  { id: 'frame-fire', name: 'Fire Avatar Frame', category: 'avatarFrames', slot: 'avatarFrame', priceCents: 299, rarity: 'common', icon: '🔥' },
  { id: 'frame-diamond', name: 'Diamond Avatar Frame', category: 'avatarFrames', slot: 'avatarFrame', priceCents: 399, rarity: 'rare', icon: '💎' },
  { id: 'frame-crown', name: 'Crown Avatar Frame', category: 'avatarFrames', slot: 'avatarFrame', priceCents: 499, rarity: 'epic', icon: '👑' },

  // Badges
  { id: 'badge-founder', name: 'Founder Badge', category: 'profileBadges', slot: 'badge', priceCents: 799, rarity: 'founder', icon: '⭐' },
  { id: 'badge-vip-monthly', name: 'VIP Monthly', category: 'vipMembership', slot: 'badge', priceCents: 1499, rarity: 'epic', grantsVip: true, icon: '👑' },
  { id: 'badge-season-pass', name: 'Season 1 Pass', category: 'vipMembership', slot: 'badge', priceCents: 1999, rarity: 'legendary', grantsVip: false, icon: '🎫' },
  { id: 'badge-hbcu-pride', name: 'HBCU Pride Badge', category: 'profileBadges', slot: 'badge', priceCents: 599, rarity: 'rare', icon: '🎓' },

  // Emotes
  { id: 'emote-all-in', name: 'All-In Emote', category: 'emotes', slot: 'emote', priceCents: 199, rarity: 'common', icon: '🤑' },
  { id: 'emote-100', name: '💯 Emote', category: 'emotes', slot: 'emote', priceCents: 199, rarity: 'common', icon: '💯' },
  { id: 'emote-chef-kiss', name: 'Chef Kiss Emote', category: 'emotes', slot: 'emote', priceCents: 299, rarity: 'rare', icon: '👨‍🍳' },
];

export const userInventory = new Map();
export const equippedCosmetics = new Map();

export const defaultOwnedCosmetics = [
  'avatar-gold-shooter',
  'avatar-royal-player',
  'avatar-neon-ace',
  'avatar-h-town-legend',
  'avatar-queen-of-clubs',
  'avatar-spade-king',
  'avatar-diamond-boss',
  'avatar-club-captain',
];

export const defaultEquippedCosmetics = {
  avatar: 'avatar-gold-shooter',
  cardBack: null,
  tableSkin: null,
  avatarFrame: null,
  badge: null,
  emote: null,
  watch: null,
  chain: null,
  ring: null,
  vehicle: null,
  home: null,
  outfit: null,
  background: null,
};

export function ensureUserCosmetics(userId) {
  if (!userInventory.has(userId)) {
    userInventory.set(userId, [...defaultOwnedCosmetics]);
  }
  if (!equippedCosmetics.has(userId)) {
    equippedCosmetics.set(userId, { ...defaultEquippedCosmetics });
  }

  return {
    inventory: userInventory.get(userId),
    equipped: equippedCosmetics.get(userId),
  };
}

sponsors.set('sponsor-001', {
  id: 'sponsor-001',
  name: 'Founders Sponsor',
  status: 'ACTIVE',
  budgetCents: 250000,
});

tournaments.set('tour-001', {
  id: 'tour-001',
  name: 'Friday Night Shooters Freeroll',
  entryFeeCents: 0,
  rebuyCents: 0,
  prizeCents: 50000,
  prizeSource: 'SPONSOR',
  sponsorId: 'sponsor-001',
  usesPlayerFunds: false,
  allowsChipCashout: false,
  status: 'OPEN',
  rulesVersion: '2026.1',
  entrants: [],
});

leagues.set('league-001', {
  id: 'league-001',
  name: 'City vs City Creator League',
  season: 'Season 1',
  sponsorId: 'sponsor-001',
  prizeCents: 100000,
  prizeSource: 'SPONSOR',
  status: 'OPEN',
  members: [],
});

leaderboards.set('league-001', []);

export const newId = (prefix) => `${prefix}-${uuid().slice(0, 8)}`;
